### PR TITLE
Fix equality check of text documents to resolve inactive region highlights not being displayed

### PR DESCRIPTION
**Pull request by kon72**
Previously, on line 100 in [vscode-clangd/src/inactive-regions.ts](https://github.com/clangd/vscode-clangd/blob/cee0726d4616d500a5989c28cbcb6a8ab08bef9f/src/inactive-regions.ts#L93-L104), the result of `vscode.Uri.toString()` was compared against `fileUri` that is a raw response from the server. This is a problem because `fileUri` is encoded on the server in a different way than vscode does in `toString()` method. For example, the server may encode a URI as `file:///C:/src/abc%23.h` while vscode will encode the same URI as `file:///C%3A/src/abc%23.h`. This causes the comparison to fail and the inactive regions are not highlighted for files that have a `:` in their path (as is always the case on Windows).

This change fixes the problem by parsing the server response into a `vscode.Uri` and then retrieving the normalized path through `fsPath`.

https://github.com/clangd/vscode-clangd/blob/cee0726d4616d500a5989c28cbcb6a8ab08bef9f/src/inactive-regions.ts#L93-L104

### CODE REVIEW

**Comment by GitMensch** [comment]
Looks good to me. The only issue: neither @HighCommander4 's original change nor that one has any testsuite entry - is it possible to add tests that use the same content with different kind of paths?

**Comment by kon72** [comment]
> is it possible to add tests that use the same content with different kind of paths?

Sure, I will add some tests.

**Review by HighCommander4 — COMMENTED** [review]
Thanks very much for adding this test suite, it's very helpful to be able to write tests for client-side features like this!

One question for future reference: do you know if there is a way to run the tests from the VSCode UI? So far the only way I've figured out is to run them on the command line using `npm run test`, but that gives `Error: Running extension tests from the command line is currently only supported if no other instance of Code is running.` unless I close all VSCode windows first, which makes for a bit of cumbersome workflow when iterating on a test.

**Comment by kon72** [comment]
> One question for future reference: do you know if there is a way to run the tests from the VSCode UI? So far the only way I've figured out is to run them on the command line using `npm run test`, but that gives `Error: Running extension tests from the command line is currently only supported if no other instance of Code is running.` unless I close all VSCode windows first, which makes for a bit of cumbersome workflow when iterating on a test.

[VSCode docs](https://code.visualstudio.com/api/working-with-extensions/testing-extension) says that there are two options to get tests work:

1. Use VS Code Insiders for development and use VS Code Stable for tests - that enforces developers to use insiders, which is a bit too high requirement in my sense.
2. Run tests from debug launch configuration (already set up in `.vscode/launch.json`) - where we can use stable version of VS Code, but cannot run tests from command line while VS Code is running...

There's another option yet not stated in the doc above:

3. Use VS Code Insiders for tests - in this way we can use stable version of VS Code for development, and we are able to run tests from command line.

Well the third option sounds flaky, but given that [even some official extensions are tested in this way](https://grep.app/search?q=version%3A%20%27insiders%27&case=true&filter[repo.pattern][0]=vscode), would this be a good option to go with?

File: test/inactive-regions.test.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    const uris = [
+      {
+        uri: vscode.Uri.file('/path/to/source.c'),
+        fromServer: 'file:///path/to/source.c'
+      },
+      {
+        // Clangd server would encode colons but `vscode.Uri.toString()`
+        // doesn't. See #515 for details.
+        uri: vscode.Uri.file('C:/path/to/source.c'),
+        fromServer: 'file:///C%3A/path/to/source.c'
```
[the remark this answers is not in the document]
Comment (kon72) [reply test/inactive-regions.test.ts]: You are right, fixed it

File: src/inactive-regions.ts [hunk 2]
```diff
@@ -83,21 +83,21 @@ export class InactiveRegionsFeature implements vscodelc.StaticFeature {
   }
 
   handleNotification(params: InactiveRegionsParams) {
-    const fileUri = params.textDocument.uri;
+    const filePath = vscode.Uri.parse(params.textDocument.uri).fsPath;
```
[the remark this answers is not in the document]
Comment (kon72) [reply src/inactive-regions.ts]: Done

**Comment by HighCommander4** [comment]
>     2. Run tests from debug launch configuration (already set up in `.vscode/launch.json`) - where we can use stable version of VS Code both for development and testing

Thanks -- this is what I was looking for. I just hadn't used the Run/Debug features of VSCode before and wasn't sure where to look, but I see it now.

I think this is good enough for now.

File: test/inactive-regions.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (HighCommander4) [reply test/inactive-regions.test.ts]: Thanks!

I think I discovered a problem with the test, though: if I change the value of `fromServer` to a completely different path, such as `'file:///completely/different/path'`, the test still passes for me. It shouldn't, should it?

[the remark this answers is not in the document]
Comment (kon72) [reply test/inactive-regions.test.ts]: After some digging, it turns out test scripts were not compiled before running tests through Run/Debug features.

Fixed build scripts and now it should be working well...

[the remark this answers is not in the document]
Comment (HighCommander4) [reply test/inactive-regions.test.ts]: Ah, good catch

**Review by HighCommander4 — APPROVED** [review]

**Comment by HighCommander4** [comment]
Thanks again for your work here @kon72!

**Pull request merged**